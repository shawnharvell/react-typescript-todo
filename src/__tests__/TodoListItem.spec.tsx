import React from "react";
import { render, act, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { TodoListItem, TodoListItemProps } from "../TodoListItem";

const inputProps: TodoListItemProps[] = [
  { todo: { complete: false, text: "item 1" }, toggleComplete: jest.fn() },
  { todo: { complete: true, text: "item 2" }, toggleComplete: jest.fn() },
];
describe("TodoListItem", () => {
  it.each(inputProps)("snapshot test", (props: TodoListItemProps) => {
    const { baseElement } = render(<TodoListItem {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  it.each(inputProps)("event handler fired as expects", (props: TodoListItemProps) => {
    render(<TodoListItem {...props} />);

    act(() => {
      userEvent.click(screen.getByLabelText(props.todo.text));
    });
    expect(props.toggleComplete).toHaveBeenCalled();
  });
});
