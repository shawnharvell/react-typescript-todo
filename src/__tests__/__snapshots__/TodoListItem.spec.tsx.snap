// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem snapshot test 1`] = `
<body>
  <div>
    <li>
      <label>
        <input
          type="checkbox"
        />
        item 1
      </label>
    </li>
  </div>
</body>
`;

exports[`TodoListItem snapshot test 2`] = `
<body>
  <div>
    <li>
      <label
        class="complete"
      >
        <input
          checked=""
          type="checkbox"
        />
        item 2
      </label>
    </li>
  </div>
</body>
`;
